src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -324,6 +342,14 @@ protected override Task WriteResponseHeadersAsyncCore(Metadata responseHeaders)
 
         public void Initialize()
         {
+            var currentActivity = Activity.Current;
+            if (currentActivity != null)
+            {
+                Activity.Current.AddTag("grpc.method", MethodCore);
```

Comment:
Who will see this? This happens within the GRPC pipeline itself and since no event is being written to the diagnostic source how will consumers get notified cc @lmolkova

Reply:
When the hosting activity is stopped I presume that OpenTelemetry can get the tag values off the activity and record them.

Reply:
If you have a request that starts out http then becomes GRPC because of the tag and is hanging for 1 minute, it won't be GRPC until that minute is up.

Reply:
Yes. Do you want a new activity for gRPC? It feels like overkill for this situation to me.

Or do activities something like `ActivityName.Update` to raise on the diagnostic source? That could be used to tell the subscriber that the metadata on the activity has changed.

Reply:
I'd lean towards a new gRPC activity. I don't think it would be too much overhead.

Reply:
A new activity seems unfortunate as well especially because it's not exactly free. @lmolkova can we abuse OnActivityImport for this?

Reply:
From the consumer side: AI or OT we'll use `OnActivityImport` to make sampling decision and it's hard to distinguish the first call that is intended for sampling from subsequent calls to update Activity state. It has to be done by using `OnActivityImport` object arg which is yet another contract between grpc and listener. 

Also for both AI and OT - we don't care much what happens with ASP.NET Core Activity until it is stopped - this is when the recording of activity happens. 

So I suggest that for now 
- we can be fine with the same Activity and just the Tag - it will be respected even though it appears after Activity starts at an unpredictable time. 
- using OnActivityImport will make default http case harder. 
- if we don't want to introduce new activity (and it's good), but has to notify someone about changes - just send another DS event like "Microsoft.AspNetCore.HttpIn.Change".

Reply:
I've updated the server to write to the diagnostic source when the activity changes:

`_diagnosticSource.Write(currentActivity.OperationName + ".Change", null);`

Reply:
Yea but don’t allocate , this does a concat every request

Reply:
I just went digging around in the open telemetry source and found this:

https://github.com/open-telemetry/opentelemetry-dotnet/blob/3538e05f0eed9df008d30e82a70bef11bd579bb5/src/OpenTelemetry.Collector.AspNetCore/Implementation/HttpInListener.cs#L108-L137

This has the exact same problem as GRPC so we should solve these similarly. 

- The request begins
- It becomes MVC after routing much like GRPC is decided after routing runs
- An event is fired that upgrades the current span

Reply:
I think we should just add an event to the route matching to handle these types of cases. The endpoint should have enough metadata to determine what type of request it is as a result.